src/plugin/nodeSpec.ts

```diff
@@ -204,8 +204,10 @@ const getDefaultParseDOMForLeafNode = (
         return false;
       }
 
+      const fields = dom.getAttribute("fields");
+
       const attrs = {
-        fields: JSON.parse(dom.getAttribute("fields") ?? "{}") as unknown,
+        fields: (fields ? JSON.parse(fields) : undefined) as unknown,
```

Comment:
I had to change the default parsing rule for a leaf node, from am empty object to undefined. This was due to issues with the dropdown being able to find the selected value on a refresh. This may have unintended consequences for other fields, I can control this with a function argument if that would be useful?

Reply:
Difficult to unravel what this might mean, as we'd be very surprised if fields weren't present on this node! Arguably this makes more sense, as an empty object doesn't make sense for many sorts of attributes (e.g. string, number); undefined is probably less surprising. Whether we account for this possibility in the types for that field is another matter. The non-presence of the `field` attribute is arguably a good spot to throw an error.

However, we should continue to use the null coalescing operator (`??`) when we're deciding whether to fall through to `undefined`, because values like `""` or `0` will be coerced to `undefined` by this change.

Reply:
Great point, I will fix that.

Reply:
Ah, specifically in the case when the attribute does not exist (not when it does exist, but contains `undefined`) – but arguably we shouldn't be throwing at this point anyhow 👍

Reply:
Yeah it does make sense, but I'm not sure how we could check for the case where it exists but is undefined. `dom.getAttribute("fields")` returns a `string | null`.